src/main/resources/assets/js/app/create/NewPrincipalDialog.ts

```diff
@@ -34,17 +36,22 @@ export class NewPrincipalDialog
     }
 
     doRender(): Q.Promise<boolean> {
-        return super.doRender().then((rendered) => {
-            this.addCancelButtonToBottom(null, true);
+        return super.doRender().then((rendered: boolean) => {
             this.getContentPanel().appendChild(this.grid);
+            const cancelButton: DialogButton = this.addCancelButtonToBottom(null, true);
+            this.setElementToFocusOnShow(cancelButton);
+            this.grid.setElementToFocusWhenNoSeleted(cancelButton);
+            this.getBody().getEl().setAttribute('tabindex', '0');
+            this.grid.setElementToFocusWhenItemSeleted(this.getBody());
```

Comment:
Good idea, but this creates too tight a coupling between the grid and the elements to be focused. Grid should for example check if any elements are set before it attempts to focus them. I suggest we do this a bit different - via events.

Whenever the grid gets or looses selection, it would fire a corresponding event and whatever container is hosting the grid can react to these events by setting focus to whatever elements it wants.

Reply:
Updated